src/adapter/src/coord/read_policy.rs

```diff
@@ -64,18 +66,113 @@ impl<T: timely::progress::Timestamp> ReadCapability<T> {
 /// Relevant information for acquiring or releasing a bundle of read holds.
 #[derive(Clone)]
 pub(crate) struct ReadHolds<T> {
-    pub(crate) time: T,
-    pub(crate) id_bundle: CollectionIdBundle,
+    holds: HashMap<Antichain<T>, CollectionIdBundle>,
```

Comment:
@frankmcsherry Could you take a look at this struct and the following methods below?
- `initialize_read_policies` which starts on line 228
- `initialize_read_holds` which starts on line 338
- `acquire_read_holds` which starts on line 388
- `update_read_hold` which starts on line 433

Another general question I have is, is it OK to use `Antichain` here as the key to a `HashMap`? I know I asked a couple days ago, and I thought that the answer was that two equal `Antichain`s may produce different hashes so this isn't safe. However, I looked at the `Antichain` impl of `Hash` and it looks like they're converted to `Vec`s and sorted first so this shouldn't be the case.

Reply:
I spoke about this with @frankmcsherry in person and applied all the recommended changes in [a41d28f](https://github.com/MaterializeInc/materialize/pull/15644/commits/a41d28f4ae081d4f51b107916c8cd344a8f9120c).